### PR TITLE
BUG: Fix unintentionally added suffix in DataFrame.apply/agg and Series.apply/agg

### LINKED ISSUE #36189: BUG: Inconsistent behavior of index values for `DataFrame.apply` and `Series.apply`
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import pandas as pd
pdf = pd.DataFrame([[4, 9]] * 3, columns=['A', 'B'])
pdf.apply(["sum", lambda df: df.sum(), lambda df: df.sum()]) # lambdas have postfix 0, 1 ...
             A   B
sum         12  27
<lambda_0>  12  27
<lambda_1>  12  27
s = pd.Series([4] * 3)
s.apply(["sum", lambda df: df.sum(), lambda df: df.sum()]) # lambdas have not postfix 0, 1 ...
sum         12
<lambda>    12
<lambda>    12
dtype: int64
```

#### Problem description

Could anyone explain please? Is it normal behavior? Why DataFrame's index has postfix for lambdas, but Series's index hasn't. The same behavior is observed for `agg`. There is no such behavior in pandas==1.0.5.

#### Output of ``pd.show_versions()``

<details>

pandas : 1.1.1
numpy : 1.18.4
pytz : 2020.1
dateutil : 2.8.1
pip : 20.1.1
setuptools : 41.2.0
Cython : None
pytest : 5.4.2
hypothesis : None
sphinx : None
blosc : None
feather : 0.4.1
xlsxwriter : None
lxml.etree : 4.5.0
html5lib : None
pymysql : None
psycopg2 : None
jinja2 : 2.11.2
IPython : 7.14.0
pandas_datareader: None
bs4 : 4.9.1
bottleneck : None
fsspec : 0.7.3
fastparquet : None
gcsfs : None
matplotlib : 3.2.1
numexpr : 2.7.1
odfpy : None
openpyxl : 3.0.3
pandas_gbq : None
pyarrow : 0.16.0
pytables : None
pyxlsb : None
s3fs : 0.4.2
scipy : 1.4.1
sqlalchemy : 1.3.17
tables : 3.6.1
tabulate : None
xarray : 0.15.1
xlrd : 1.2.0
xlwt : None
numba : None

</details>

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [x] closes #36189
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.1.3.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@ including other versions of pandas.
 
 Fixed regressions
 ~~~~~~~~~~~~~~~~~
--
+- Regression in :meth:``DataFrame.agg``, :meth:``DataFrame.apply``, :meth:``Series.agg``, and :meth:``Series.apply`` where internal suffix is exposed to the users when no relabelling is applied. (:issue:`36189`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.1.3.rst:17]: only single backticks after `:meth:`

Comment (charlesdong1991) [reply doc/source/whatsnew/v1.1.3.rst]: thanks, changed!

**Review by jreback — APPROVED** [review]
very nice @charlesdong1991 merge on green.

**Comment by simonjayhawkins** [comment]
ci restarted

=========================== short test summary info ===========================
FAILED pandas/tests/plotting/test_groupby.py::TestDataFrameGroupByPlots::test_series_groupby_plotting_nominally_works
= 1 failed, 75150 passed, 1789 skipped, 1180 xfailed, 6 xpassed, 69 warnings in 1028.46s (0:17:08) =

**Comment by simonjayhawkins** [comment]
Thanks @charlesdong1991

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Comment by charlesdong1991** [comment]
thank you @simonjayhawkins for pushing this through yesterday! 🙇 

> =========================== short test summary info ===========================
FAILED pandas/tests/plotting/test_groupby.py::TestDataFrameGroupByPlots::test_series_groupby_plotting_nominally_works
= 1 failed, 75150 passed, 1789 skipped, 1180 xfailed, 6 xpassed, 69 warnings in 1028.46s (0:17:08) =

May I ask where was this failure popping up? is it something I should take a look for a potential issue? @simonjayhawkins

**Comment by simonjayhawkins** [comment]
not sure. I just add the test failure to increase visibility and help track. in this case has occurred b4 https://github.com/pandas-dev/pandas/pull/36147#issuecomment-688883534

**Comment by simonjayhawkins** [comment]
failing case is here https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=42259&view=logs&jobId=a3a13ea8-7cf0-5bdb-71bb-6ac8830ae35c&j=4b4d380c-92b5-58cf-5b80-2c6c9b1d5805&t=8eab1181-1335-58ed-7ed5-a271d9d86005&s=d654deb9-056d-50a2-1717-90c08683d50a

**Pull request merged**